configmicrok8s.sh

```diff
@@ -1,16 +1,211 @@
+sleep 3 
+
+flux create secret git gitops -n flux-system \
+--url "https://github.com/$GITOPS_REPO" \
+--password "$GITOPS_PAT" \
+--username gitops
+
+kubectl apply -k "clusters/$STORE_NAME/flux-system/flux-system" 
+sleep 5
+
+flux reconcile source git gitops
+printf '\n Flux installed successfully ✅\n'
+
+# Switching Back to Home Directory 
+cd $HOME
+
+##### ARC region ######
+
+printf "\n Logging in Azure using Service Principal 🚧 \n"
+# Az Login using SP
+az login --service-principal -u $AZ_SP_ID  -p  $AZ_SP_SECRET --tenant $AZ_TEANANT_ID
+
+# Arc setup 
+az extension add --name connectedk8s
+
+az provider register --namespace Microsoft.Kubernetes
+az provider register --namespace Microsoft.KubernetesConfiguration
+az provider register --namespace Microsoft.ExtendedLocation
+
+# Check for existing resource group
+if [ $(az group exists --name $AZ_ARC_RESOURCEGROUP) == false ]; then
+    az group create --name $AZ_ARC_RESOURCEGROUP --location $AZ_ARC_RESOURCEGROUP_LOCATION --output table
+    printf "\n Resource group $AZ_ARC_RESOURCEGROUP created ✅\n"
+fi
+
+printf "\n Connecting to Azure Arc 🚧 \n"
+az connectedk8s connect --name $STORE_NAME --resource-group $AZ_ARC_RESOURCEGROUP
+
+printf "\n Creating k8s-extension 🚧 \n"
+az k8s-extension create --cluster-name $STORE_NAME --resource-group $AZ_ARC_RESOURCEGROUP --cluster-type connectedClusters --extension-type Microsoft.AzureKeyVaultSecretsProvider --name $SECRET_PROVIDER_NAME
+
+printf "\n Verifying k8s-extension 🚧 \n"
+az k8s-extension show --cluster-type connectedClusters --cluster-name $STORE_NAME --resource-group $AZ_ARC_RESOURCEGROUP --name $SECRET_PROVIDER_NAME
+
+
+#TODO: Check if service account already present
+# Generate token to connect to Azure k8s cluster
+ADMIN_USER=$(kubectl get serviceaccount admin-user -o jsonpath='{$.metadata.name}' --ignore-not-found)
+if [ -z "$ADMIN_USER" ]; then
+    printf "\n Creating service account 🚧 \n"
+    kubectl create serviceaccount admin-user
+else
+    printf "\n Service account already exist. \n"
+fi
+
+CLUSTER_ROLE_BINDING=$(kubectl get clusterrolebinding admin-user-binding -o jsonpath='{$.metadata.name}' --ignore-not-found)
+if [ -z "$CLUSTER_ROLE_BINDING" ]; then
+    printf "\n Creating cluster role binding 🚧 \n"
+    kubectl create clusterrolebinding admin-user-binding --clusterrole cluster-admin --serviceaccount default:admin-user
+else
+    printf "\n Cluster role binding already exist. \n"     
+fi
+
+# Generating a secret
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: admin-user
+  annotations:
+    kubernetes.io/service-account.name: admin-user
+type: kubernetes.io/service-account-token
+EOF
+
+TOKEN=$(kubectl get secret admin-user -o jsonpath='{$.data.token}' | base64 -d | sed $'s/$/\\\n/g')
+
+printf "\n ####### Token to connect to Azure ARC starts here ######## \n"
+printf $TOKEN
+printf "\n ####### Token to connect to Azure ARC ends here   ######### \n"
```

Comment:
is it necessary to write the token to disk in clear text?

If so, put in the .ssh directory and chmod to min permissions.